### ISSUE TITLE
NOT NULL on uuid columns (Sideeffects and possible defect)

### ISSUE THREAD

**Issue by toirl** [ringo-framework/ringo#97]
Commit b04ef3f2d70950829b8012046d701efaafd4fd40 of PR #87 by @g-tom introcduced a not null constraint on uuid column of all tables. This is generally an improvement to enforce a value. Unfortunately this introduces the problem that it is not possible anymore to write  INSERT statements to insert data into the database without generating a new UUID before. This wasn't considered beeing a problem during the review of the PR, including myself.

This problem affects at least the migrations. I expect some of the ringo-admin commands will also have issues with this (e.g add modul, which inserts new entries in the modul table), but I did not have tested this so far. 

At least for the migrations a workaround might be to generate some uuid on the python shell and use those UUID in the migrations:

`
import uuid
for u in range(10):
    str(uuid.uuid4())
`

However. It would be nice if we can make the DB autogenerate a UUID in case it is missing.

**Comment by g-tom** [comment]
> Unfortunately this introduces the problem that it is not possible anymore to write INSERT statements to insert data into the database without generating a new UUID before.

This is correct in case the statement is not generated by the ORM. Otherwise the default already solves the problem:

https://github.com/ringo-framework/ringo/blob/2f84645b2ddd57dac3fa8f172552b08fe186b97d/ringo/model/base.py#L145-L148

> It would be nice if we can make the DB autogenerate a UUID in case it is missing.

You can have a look at the implementation in AGrav (unfortunately not publicly available), if you really need the UUIDs generated by the database. It's based on pgcrypto:
https://www.postgresql.org/docs/10/static/pgcrypto.html#idm45262034210592

**Comment by g-tom** [comment]
Because of the title here: see also #98
